public/app/features/dashboard/services/PublicDashboardDataSource.ts

```diff
@@ -0,0 +1,67 @@
+import { DataQuery, DataQueryRequest, DataQueryResponse, DataSourceApi, PluginMeta } from '@grafana/data';
+import { BackendDataSourceResponse, getBackendSrv, toDataQueryResponse } from '@grafana/runtime';
+import { catchError, Observable, of, switchMap } from 'rxjs';
+
+export class PublicDashboardDataSource extends DataSourceApi<any> {
```

Comment:
This seems awkward -- is the idea that we want different instances that only hit the "public" endpoint?

It seems much more valuable to just extend `DataSourceWithBackend` so that we can use the "public" endpoint when necessary and the saved path when we can.